lib/logstash/codecs/cef.rb

```diff
@@ -103,14 +103,31 @@ def decode(data)
         message=message + ' ' unless message.end_with?('\=')
       end
 
-      # Now parse the key value pairs into it
-      extensions = {}
-      message = message.split(/ ([\w\.]+)=/)
-      key, value = message.shift.split('=', 2)
-      extensions[key] = value.gsub(/\\=/, '=').gsub(/\\\\/, '\\')
-      Hash[*message].each{ |k, v| extensions[k] = v }
-      # And save the new has as the extensions
-      event.set('cef_ext', extensions)
+      # Introducing custom delimiters instead of space to avoid convolution when parsing fields with ' ' spaces.
+      # Example : cs1Label=New Malware cs1=Conficker ==> cs1Label=New Malware|^^^cs1=Conficker
+      message=message.gsub((/\s+(\w+=)/)||(/(\w+=)/),'|^^^\1')
+
+      # Truncating the additional fields, if needed it can be mapped from ArcSight's connector; otherwise also protects the dot (.) notations in implmentations for Elasticsearch v 2.4 and below
+      # Example ad.field[0]=XXXX ad.name[2]=abc
+      message=message.gsub((/ (ad.\w+.*\]\=[^|^^^]+)/),'')
```

Comment:
the regex `/ad.` will match "a", "d", followed by any character. Based on your comment about the dot notation and the example, you probably want to escape the dot `\.`